packages/wallet/tests/mocks/SecureSite.mock.ts

```diff
@@ -0,0 +1,29 @@
+export const SecureSiteMock = {
+  approveRequest: ({ id, type, response }: { id: string; type: string; response: unknown }) =>
+    setTimeout(() => {
```

Comment:
https://walletconnect.slack.com/archives/C03RVH94K5K/p1721118911888149?thread_ts=1721083113.218169&cid=C03RVH94K5K

Was is required to use `setTimeout` and why?

Reply:
This is to simluate Promise like behavior (for typing) but I changed it to use `Promise.resolve()` instead since it's more clear 👌